vm/src/builtins/set.rs

```diff
@@ -573,23 +572,26 @@ impl PyFrozenSet {
         iterable: OptionalArg<PyObjectRef>,
         vm: &VirtualMachine,
     ) -> PyResult<PyRef<Self>> {
-        let iterable = if let Present(iterable) = iterable {
-            if cls.is(&vm.ctx.types.frozenset_type) {
-                match iterable.downcast_exact::<PyFrozenSet>(vm) {
-                    Ok(iter) => return Ok(iter),
-                    Err(iterable) => Present(PyIterable::try_from_object(vm, iterable)?),
+        let elements = if let OptionalArg::Present(iterable) = iterable {
+            let iterable = if cls.is(&vm.ctx.types.frozenset_type) {
```

Comment:
Though I've seen this used in other `__new__`'s it seems superfluous. Wouldn't:

```rust
let iterable = match iterable.downcast_exact::<Self>(vm) {
    Ok(fs) => return Ok(fs),
    Err(iterable) => iterable,
};
```

suffice in only allowing `frozenset` instances (and not subclasses of `frozenset`!) to return immediately?